apm-agent-plugins/apm-jdbc-plugin/src/main/java/co/elastic/apm/agent/jdbc/JdbcInstrumentation.java

```diff
@@ -26,18 +26,32 @@
 
 import co.elastic.apm.agent.bci.TracerAwareInstrumentation;
 import co.elastic.apm.agent.jdbc.helper.JdbcHelper;
+import net.bytebuddy.matcher.ElementMatcher;
 
+import javax.annotation.Nullable;
 import java.util.Collection;
 import java.util.Collections;
 
+import static co.elastic.apm.agent.bci.bytebuddy.CustomElementMatchers.classLoaderCanLoadClass;
+
 public abstract class JdbcInstrumentation extends TracerAwareInstrumentation {
 
     private static final Collection<String> JDBC_GROUPS = Collections.singleton("jdbc");
 
-    protected static JdbcHelper jdbcHelper = new JdbcHelper();
+    @Nullable
+    private static JdbcHelper jdbcHelper;
 
     @Override
     public final Collection<String> getInstrumentationGroupNames() {
         return JDBC_GROUPS;
     }
+
+    protected synchronized static JdbcHelper getJdbcHelper() {
+        // lazily initialize helper to prevent trying to load classes in java.sql package with the bootstrap classloader
```

Comment:
Is that an indication that we should more strictly separate advices from element matches?

Reply:
I would say yes, separation would be better. In the same category, there is `getAdviceClass()` method that returns the advice class loaded from agent/bootstrap classloader when in practice we only use class name and then load it in child-first plugin classloader. We already discussed that a bit with @eyalkoren last week.